hail_scripts/utils/clinvar.py

```diff
@@ -21,9 +36,67 @@
         "practice_guideline": 4,
     }
 )
-
-
-def download_and_import_latest_clinvar_vcf(genome_version: str) -> hl.MatrixTable:
+# NB: sorted by pathogenicity
+CLINVAR_PATHOGENICITIES = [
+    'Pathogenic',
+    'Pathogenic/Likely_pathogenic',
+    'Pathogenic/Likely_pathogenic/Likely_risk_allele',
+    'Pathogenic/Likely_risk_allele',
+    'Likely_pathogenic',
+    'Likely_pathogenic/Likely_risk_allele',
+    'Established_risk_allele',
+    'Likely_risk_allele',
+    'Conflicting_interpretations_of_pathogenicity',
+    'Uncertain_risk_allele',
+    'Uncertain_significance/Uncertain_risk_allele',
+    'Uncertain_significance',
+    CLINVAR_DEFAULT_PATHOGENICITY,
+    'Likely_benign',
+    'Benign/Likely_benign',
+    'Benign',
+]
+CLINVAR_PATHOGENICITIES_LOOKUP = hl.dict(hl.enumerate(CLINVAR_PATHOGENICITIES, index_first=False))
+
+def parsed_clnsig(ht: hl.Table):
+    return (
+        hl.delimit(ht.info.CLNSIG)
+        .replace(
+            'Likely_pathogenic,_low_penetrance', 'Likely_pathogenic|low_penetrance',
+        )
+        .replace(
+            '/Pathogenic,_low_penetrance', '|low_penetrance',
+        )
+        .split(r'\|')
+    )
+
+def parsed_clnsigconf(ht: hl.Table):
+
+    def parse_to_count(entry: str):
+        splt = entry.split(r'\(') # pattern, count = entry... if destructuring worked on a hail expression!
+        return hl.tuple([
+            splt[0], 
+            hl.int32(splt[1][:-1])
+        ])
+
+    return (
+        hl.delimit(ht.info.CLNSIGCONF)
+        .replace(',_low_penetrance', '')
+        .split(r'\|')
+        .map(parse_to_count)
+        .group_by(lambda x: x[0])
+        .map_values(lambda values: (
+            values.fold(
+                lambda x, y: x + y[1],
+                0,
+            )
+        ))
+        .items()
+        .map(
+            lambda x: hl.Struct(pathogenicity_id=CLINVAR_PATHOGENICITIES_LOOKUP[x[0]], count=x[1])
```

Comment:
This works, but I think the whole thing might be more readable if `parse_to_count` returned a struct in the first place.For example, I think that `.group_by(lambda x: x. pathogenicity_id)` is more readable than `.group_by(lambda x: x[0])`, etc.